src/basecamp.cpp

```diff
@@ -804,6 +804,23 @@ void basecamp::unload_camp_map()
     }
 }
 
+void basecamp::set_owner( faction_id new_owner )
+{
+    for( const std::pair<faction_id, faction> fac : g->faction_manager_ptr->all() ) {
+        if( fac.first == new_owner ) {
+            owner = new_owner;
+            return;
+        }
+    }
+    //Fallthrough, id must be invalid
+    debugmsg( "Could not find matching faction for new owner's faction_id!" );
+}
+
+faction_id basecamp::get_owner()
```

Comment:
```suggestion
faction_id basecamp::get_owner() const
```
Shouldn't this be `const` or it doesn't matter?

Reply:
From a purely technical perspective it *might* allow the slimmest of compiler optimizations, but const-qualifying the function would not provide any of the normal protection (inability to modify members of the parent object) because `owner` is intentionally `mutable`.